toolflow/vivado/platform/xupvvh/plugins/sfpplus_100g.tcl

```diff
@@ -0,0 +1,177 @@
+        set name [dict get $ports $port]
+        generate_core $port $name $first_port $constraints_file
+        incr first_port 1
+      }
+
+      close $constraints_file
+      read_xdc $constraints_fn
+      set_property PROCESSING_ORDER NORMAL [get_files $constraints_fn]
+    }
+
+    # Generate a SFP+-Core to handle the ports of one physical cage
+    # @param physical_port the number of the physical cage
+    # @param name name of the port
+    # @param first_port the first free master on the AXI-Lite Config interconnect
+    # @param constraints_file the file used for constraints
+    proc generate_core {physical_port name first_port constraints_file} {
+      variable refclk_pins
+      variable cmac_cores
+      variable gt_groups
+      variable qsfp_rst
+
+      set port_qsfp_rst [create_bd_port -dir O qsfp_rst_l_$physical_port]
+      puts $constraints_file [format {set_property PACKAGE_PIN %s [get_ports %s]} [lindex $qsfp_rst $physical_port] qsfp_rst_l_$physical_port]
+      puts $constraints_file [format {set_property IOSTANDARD LVCMOS18 [get_ports %s]} qsfp_rst_l_$physical_port]
+
+      # Create and constrain refclk pin
+      set gt_refclk [create_bd_intf_port -mode Slave -vlnv xilinx.com:interface:diff_clock_rtl:1.0 gt_refclk_$physical_port]
+      set_property CONFIG.FREQ_HZ 322265625 $gt_refclk
+      puts $constraints_file [format {set_property PACKAGE_PIN %s [get_ports %s]} [lindex $refclk_pins $physical_port] gt_refclk_${physical_port}_clk_p]
+
+      # Create and configure core
+      set core [tapasco::ip::create_100g_ethernet ethernet_$physical_port]
+
+      set_property -dict [list \
+        CONFIG.CMAC_CAUI4_MODE {1} \
+        CONFIG.NUM_LANES {4x25} \
+        CONFIG.USER_INTERFACE {AXIS} \
+        CONFIG.GT_REF_CLK_FREQ {322.265625} \
+        CONFIG.TX_FLOW_CONTROL {0} \
+        CONFIG.RX_FLOW_CONTROL {0} \
+        CONFIG.INCLUDE_RS_FEC {1} \
+        CONFIG.ENABLE_AXI_INTERFACE {0} \
+        CONFIG.INCLUDE_STATISTICS_COUNTERS {0} \
+        CONFIG.RX_MAX_PACKET_LEN {16383} \
+        CONFIG.CMAC_CORE_SELECT [lindex $cmac_cores $physical_port] \
+        CONFIG.GT_GROUP_SELECT [lindex $gt_groups $physical_port]
+        ] $core
+
+      connect_bd_intf_net $gt_refclk [get_bd_intf_pins $core/gt_ref_clk]
+      connect_bd_net [get_bd_pins $core/sys_reset] [get_bd_pins dclk_reset/peripheral_reset]
+      make_bd_intf_pins_external [get_bd_intf_pins $core/gt_rx]
+      make_bd_intf_pins_external [get_bd_intf_pins $core/gt_tx]
+      connect_bd_net [get_bd_pins $core/drp_clk] [get_bd_pins dclk_wiz/clk_out1]
+      connect_bd_net [get_bd_pins $core/init_clk] [get_bd_pins dclk_wiz/clk_out1]
+
+      # Connect core
+      connect_bd_intf_net [get_bd_intf_pins $core/axis_rx] [get_bd_intf_pins AXIS_RX_${name}]
+      connect_bd_intf_net [get_bd_intf_pins $core/axis_tx] [get_bd_intf_pins AXIS_TX_${name}]
+      #connect_bd_net [get_bd_pins $core/s_axi_aclk] [get_bd_pins dclk_wiz/clk_out1]
```

Comment:
These lines are not necessary?
So they can be removed?

Reply:
will fix